samples/filter-features-in-scene/src/main/java/com/esri/arcgismaps/sample/filterfeaturesinscene/components/FilterFeaturesInSceneViewModel.kt

```diff
@@ -0,0 +1,133 @@
+
+import android.app.Application
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.lifecycle.AndroidViewModel
+import com.arcgismaps.Color
+import com.arcgismaps.geometry.Point
+import com.arcgismaps.geometry.Polygon
+import com.arcgismaps.geometry.PolygonBuilder
+import com.arcgismaps.mapping.ArcGISScene
+import com.arcgismaps.mapping.ArcGISTiledElevationSource
+import com.arcgismaps.mapping.BasemapStyle
+import com.arcgismaps.mapping.PortalItem
+import com.arcgismaps.mapping.Viewpoint
+import com.arcgismaps.mapping.layers.ArcGISSceneLayer
+import com.arcgismaps.mapping.layers.SceneLayerPolygonFilter
+import com.arcgismaps.mapping.layers.SceneLayerPolygonFilterSpatialRelationship
+import com.arcgismaps.mapping.symbology.SimpleFillSymbol
+import com.arcgismaps.mapping.symbology.SimpleFillSymbolStyle
+import com.arcgismaps.mapping.symbology.SimpleLineSymbol
+import com.arcgismaps.mapping.symbology.SimpleLineSymbolStyle
+import com.arcgismaps.mapping.view.Camera
+import com.arcgismaps.mapping.view.Graphic
+import com.arcgismaps.mapping.view.GraphicsOverlay
+import com.arcgismaps.portal.Portal
+import com.arcgismaps.toolkit.geoviewcompose.SceneViewProxy
+import com.esri.arcgismaps.sample.sampleslib.components.MessageDialogViewModel
+import kotlinx.coroutines.runBlocking
+
+class FilterFeaturesInSceneViewModel(application: Application) : AndroidViewModel(application) {
+
+    val sceneViewProxy = SceneViewProxy()
+
+    // create a ViewModel to handle dialog interactions
+    val messageDialogVM: MessageDialogViewModel = MessageDialogViewModel()
+
+    // Create an OSM Buildings ArcGISSceneLayer from a portal item
+    private val osmBuildingsSceneLayer = ArcGISSceneLayer(
+        PortalItem(
+            portal = Portal("https://www.arcgis.com"), itemId = "ca0470dbbddb4db28bad74ed39949e25"
+        )
+    )
+
+    // Create a San Francisco Buildings ArcGISSceneLayer from a url
+    private val sanFranciscoBuildingsSceneLayer =
+        ArcGISSceneLayer(uri = "https://tiles.arcgis.com/tiles/z2tnIkrLQ2BRzr6P/arcgis/rest/services/SanFrancisco_Bldgs/SceneServer")
+
+    // Create a new ArcGISScene and set a basemap from a portal item with a vector tile layer
+    val arcGISScene: ArcGISScene by mutableStateOf(ArcGISScene(BasemapStyle.ArcGISTopographic).apply {
+        // Add an elevation source to the scene's base surface.
+        baseSurface.elevationSources.add(ArcGISTiledElevationSource(uri = "https://elevation3d.arcgis.com/arcgis/rest/services/WorldElevation3D/Terrain3D/ImageServer"))
+        // OSM building layer
+        operationalLayers.add(osmBuildingsSceneLayer)
+        // Add the buildings scene layer to the operational layers
+        operationalLayers.add(sanFranciscoBuildingsSceneLayer)
+        // Set the initial viewpoint of the scene
+        initialViewpoint = Viewpoint(
+            Point(0.0, 0.0), Camera(
```

Comment:
Not sure if this was the intention but currently, this point is inferred as a `boundingGeometry`, and if you were to define a scale the SDK would infer this point as a `center` point.

Reply:
I just find this API really odd. Since the value doesn't have an affect if a camera is passed. Anyways I've updated to something more plausible.